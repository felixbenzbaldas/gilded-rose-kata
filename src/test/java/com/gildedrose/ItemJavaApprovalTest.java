package com.gildedrose;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.util.List;

public class ItemJavaApprovalTest {

    @Test
    public void javvApprovalOneDay() {
        Item agedBrie = new Item("Aged Brie", 2, 0);
        List<Item> items = List.of(
            new Item("+5 Dexterity Vest", 10, 20),
            agedBrie,
            new Item("Elixir of the Mongoose", 5, 7),
            new Item("Sulfuras, Hand of Ragnaros", 0, 80),
            new Item("Sulfuras, Hand of Ragnaros", -1, 80),
            new Item("Backstage passes to a TAFKAL80ETC concert", 15, 20),
            new Item("Backstage passes to a TAFKAL80ETC concert", 10, 49),
            new Item("Backstage passes to a TAFKAL80ETC concert", 5, 49)
        );

        Item.nextDay(items);

        Approvals.verifyAll("Item", items);
    }
}
