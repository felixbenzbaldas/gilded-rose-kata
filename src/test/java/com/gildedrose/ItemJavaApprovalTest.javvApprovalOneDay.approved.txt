Item[0] = +5 Dexterity Vest, 9, 19
Item[1] = Aged Brie, 1, 1
Item[2] = Elixir of the Mongoose, 4, 6
Item[3] = Sulfuras, Hand of Ragnaros, 0, 80
Item[4] = Sulfuras, Hand of Ragnaros, -1, 80
Item[5] = Backstage passes to a TAFKAL80ETC concert, 14, 21
Item[6] = Backstage passes to a TAFKAL80ETC concert, 9, 50
Item[7] = Backstage passes to a TAFKAL80ETC concert, 4, 50
